frontend/gateway/client/result.go

```diff
@@ -22,20 +37,29 @@ func NewResult() *Result {
 
 func (r *Result) AddMeta(k string, v []byte) {
 	r.mu.Lock()
+	defer r.mu.Unlock()
```

Comment:
FYI, at least historically there used to be a performance penalty for running `defer` so these were not quite accidental. Still of course keep using defer if it makes code a lot cleaner. I have not checked if the performance issue has been improved in later Go versions.

Reply:
Let's stick with the old versions for now, and clean them up if/when we do a performance analysis :+1: